src/components/ebay-radio/index.js

```diff
@@ -0,0 +1,29 @@
+const markoWidgets = require('marko-widgets');
+const emitAndFire = require('../../common/emit-and-fire');
+const processHtmlAttributes = require('../../common/html-attributes');
+const template = require('./template.marko');
+
+function getInitialState(input) {
+    return {
+        classes: ['radio', input.class],
+        disabled: Boolean(input.disabled),
+        htmlAttributes: processHtmlAttributes(input)
+    };
+}
+
+function getTemplateData(state) {
+    return state;
+}
+
+function handleClick(originalEvent) {
+    if (!this.state.disabled) {
+        emitAndFire(this, 'radio-change', { originalEvent, selected: originalEvent.target.value });
```

Comment:
Will event fire even if radio is currently checked? I don't think it should, right? If so we'll need a test case for this.

Reply:
yep there is a [testcase](https://github.com/eBay/ebayui-core/pull/134/files#diff-d52de5f96198a93fd0388bde6d4395beR39) for it.

Reply:
That test says "given radio button is disabled", but I'm talking about *checked* state, not disabled state.

Reply:
ohk. Yes, the event will fire even if the radio is checked. It follows the same behavior as native [demo](http://jsbin.com/jaxifuvige/edit?html,js,console,output)

Reply:
That native demo is firing a click event though. Not a change event.